tests/conftest.py

```diff
@@ -2,22 +2,109 @@
+from indexd.index.drivers.alchemy import SQLAlchemyIndexDriver
+from indexd.alias.drivers.alchemy import SQLAlchemyAliasDriver
+from indexd.auth.drivers.alchemy import SQLAlchemyAuthDriver
+
+POSTGRES_CONNECTION = "postgresql://postgres:postgres@localhost:5432/indexd_tests"  # pragma: allowlist secret
+
+
+def clear_database():
+    """
+    Clean up test data from unit test
+    """
+    engine = create_engine(POSTGRES_CONNECTION)
+
+    with engine.connect() as conn:
+        index_driver = SQLAlchemyIndexDriver(POSTGRES_CONNECTION)
+        # IndexD table needs to be delete in this order to avoid foreign key constraint error
+        table_delete_order = [
+            "index_record_url_metadata",
+            "index_record_url",
+            "index_record_hash",
+            "index_record_authz",
+            "index_record_ace",
+            "index_record_alias",
+            "index_record_metadata",
+            "alias_record_hash",
+            "alias_record_host_authority",
+            "alias_record",
+            "index_record",
+            "drs_bundle_record",
+            "base_version",
+            "record",
+        ]
+
+        for table_name in table_delete_order:
+            delete_statement = f"DELETE FROM {table_name}"
+            conn.execute(delete_statement)
+
+        # Clear the Alias records
+        alias_driver = SQLAlchemyAliasDriver(POSTGRES_CONNECTION)
+        for model in alias_base.__subclasses__():
+            table = model.__table__
+            delete_statement = table.delete()
+            conn.execute(delete_statement)
+
+        # Clear the Auth records
+        auth_driver = SQLAlchemyAuthDriver(POSTGRES_CONNECTION)
+        for model in auth_base.__subclasses__():
+            table = model.__table__
+            delete_statement = table.delete()
+            conn.execute(delete_statement)
+
+
+@pytest.fixture(scope="function", params=["default_settings", "single_table_settings"])
+def combined_default_and_single_table_settings(request):
+    """
+    Fixture to run a unit test with both multi-table and single-table driver
+    """
+
+    # Load the default settings
+    if request.param == "default_settings":
+        from indexd import default_settings
+        from tests import default_test_settings
+
+        importlib.reload(default_settings)
+        default_settings.settings = {
+            **default_settings.settings,
+            **default_test_settings.settings,
+        }
+        yield get_app(default_settings.settings)
+
+    # Load the single-table settings
+    elif request.param == "single_table_settings":
+        from indexd import single_table_settings
+
+        importlib.reload(single_table_settings)
+        single_table_settings.settings = {
+            **single_table_settings.settings,
+        }
+        yield get_app(single_table_settings.settings)
+
+    try:
+        clear_database()
+    except Exception:
+        pass
```

Comment:
we might want to log the exception instead of silently passing (same in `app` and `user`)